File: 1041-robot-bounded-in-circle/1041-robot-bounded-in-circle.ts
function isRobotBounded(instructions: string): boolean {
    const dirs = ["N","E", "S", "W"]

    const x_0 = 0;
    const y_0 = 0;
    const dir_0 = "N";

    let x_1 = x_0;
    let y_1 = y_0;
    let dir_1 = dir_0;

    let move= "";
    let res= false;

        for(let i=0; i<instructions.length; i++){
            let indx = dirs.indexOf(dir_1);
            if(instructions[i] === "R"){
                if(indx === 3){
                    indx = 0;
                }else{
                    indx= indx+1;
                }
                dir_1=dirs[indx];
            }else{
                if(instructions[i] === "L"){
                    if(indx ===0){
                        dir_1= dirs[dirs.length-1];
                    }else{
                        indx= indx-1;
                        dir_1= dirs[indx];
                    }
                }else{
                    if(dir_1==="E"){
                        x_1++;
                    }else{
                        if(dir_1==="W"){
                            x_1--;
                        }else{
                            if(dir_1==="S"){
                                y_1--;
                            }else{
                                y_1++;
                            }
                        }
                    }
                }
            }
        }
        if((x_1===0 && y_1===0)|| dir_1!="N"){
            res= true;
            return res;
        };
    
    return res;
};

File: 1041-robot-bounded-in-circle/1041-robot-bounded-in-circle.cpp
class Solution {
public:
    bool isRobotBounded(string instructions) {
        vector<char> dirs = {'N','E', 'S', 'W'};

        int x_0 = 0;
        int y_0 = 0;
        char dir_0 = 'N';

        int x_1 = x_0;
        int y_1 = y_0;
        char dir_1 = dir_0;
        bool res= false;

        for(int i=0; i<instructions.size(); i++){
            auto it = find(dirs.begin(), dirs.end(), dir_1);
            int indx = it - dirs.begin();
            switch(instructions[i]){
                case 'R':
                    indx = (indx+1) % dirs.size();
                    dir_1=dirs[indx];
                    break;
                case 'L':
                    indx= (indx + dirs.size() - 1) % dirs.size();
                    dir_1= dirs[indx];
                    
                    break;
                case 'G':
                    if(dir_1=='E'){
                        x_1++;
                    }else{
                        if(dir_1=='W'){
                            x_1--;
                        }else{
                            if(dir_1=='S'){
                                y_1--;
                            }else{
                                y_1++;
                            }
                        }
                    }
                    break;
                default:
                    break;
                
            }
        }
        if((x_1==0 && y_1==0)|| dir_1!='N'){
            res= true;
            return res;
        };
    
    return res;
    }
};